compiler/frontend/src/hir.rs

```diff
@@ -133,71 +134,125 @@ impl Body {
 #[derive(Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct Id {
     pub module: Module,
-    pub keys: Vec<IdKey>,
+    pub keys: IdPath,
 }
+#[derive(Clone, Eq, From, Hash, Ord, PartialEq, PartialOrd, Debug)]
+pub struct IdPath(String);
 #[derive(Clone, Eq, From, Hash, Ord, PartialEq, PartialOrd)]
 pub enum IdKey {
     Named { name: String, disambiguator: usize },
     Positional(usize),
 }
+impl IdPath {
+    const fn empty() -> Self {
+        Self(String::new())
+    }
+
+    fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    fn push(&self, key: IdKey) -> Self {
+        if self.is_empty() {
+            Self::from(key)
+        } else {
+            Self(format!("{}:{key}", self.0))
+        }
+    }
+
+    #[must_use]
+    pub fn last_as_string(&self) -> Option<&str> {
+        self.0.rfind(':').map(|i| &self.0[(i + ':'.len_utf8())..])
```

Comment:
This might work as well (not tested):

```suggestion
        self.0.rsplit(':').next()
```